### ISSUE TITLE
ImageInfo`s size small then origion pic

### ISSUE THREAD

**Issue by wangfaxi1985** [facebook/fresco#1896]
When i use:
![image](https://user-images.githubusercontent.com/5296419/30268755-50d5507c-9719-11e7-81d7-82330f56614d.png)

when onFinalImageSet exe,the imageinfo`s size small then origion pic 
my origion pic`s size is :828*9164
imageinfo size is :165*1832

Fresco version:1.5
user this lib to support drag and scale Image:https://github.com/ongakuer/PhotoDraweeView

**Comment by wangfaxi1985** [comment]
![image](https://user-images.githubusercontent.com/5296419/30268839-9e6086ae-9719-11e7-9d9c-93401feccd46.png)

this is the　effect　！！！，anyone help?

**Comment by erikandre** [comment]
Hi @wangfaxi1985, when you say image size, is that width * height?
Do you get the same problem for an image with less long aspect ratio?
It would also be helpful if you could share the problematic image with us. Thanks!

**Comment by wangfaxi1985** [comment]
yes，
the image uri is:[http://oss.kitalive.tv/1814315266780168200?Expires=1505443983&OSSAccessKeyId=LTAIJ9IsE3t7IQ91&Signature=uIE5ga5afy8XD5Ry88hZT6KXZwY%3D](url)
its size is:`828*9164`，but fresco 's BaseControllerListener return 's imagesize is:`165*1832`
i use another size `3120*4160`,the fresco return size is:`780*1040`
i use image size:`200*200`,the fresco return size is ok (`200*200`)
it seem like the fresco scale imagesize when pic is big then phone aspect ratio!!!

any idea for return origion size or origion bitmap for me ?
thank fou answer me

**Comment by wangfaxi1985** [comment]
the image url may not access,this is origion pic 
![1814315266780168200](https://user-images.githubusercontent.com/5296419/30410324-f3846d7e-993c-11e7-9ae5-1613cc561736.png)

**Comment by BALUSANGEM** [comment]
@wangfaxi1985  One possibility is that you might be _enabled down sampling_ when initializing fresco

```
ImagePipelineConfig config = ImagePipelineConfig
                .newBuilder(getApplicationContext())
                .setDownsampleEnabled(true).build();
Fresco.initialize(getApplicationContext(),config);
```

**Comment by erikandre** [comment]
@wangfaxi1985 Depending on the device you are running this on, there are limits to how large a bitmap can be. On older devices it is 2048x2048 and on newer devices it will be bigger than that.

As a workaround, could you try disabling hardware acceleration for the view?

`view.setLayerType(View.LAYER_TYPE_SOFTWARE, null)`

**Comment by wangfaxi1985** [comment]
@erikandre ,i try set  `view.setLayerType(View.LAYER_TYPE_SOFTWARE, null)`, but not work ,tks
@LUSANGEM  yes i set `setDownsampleEnabled(true)`,is other way to close `setDownsampleEnabled(false)` for temporary

**Comment by erikandre** [comment]
@wangfaxi1985 - the next should contain support for region decoding of images. That would allow you to only load a part of the image. That does not really solve the problem, but if support for this was added to PhotoDraweeView then you could have a smaller version when zoomed out, and then load a higher resolution version when zooming in.

**Comment by oprisnik** [comment]
A really basic example for this is now available in the Showcase app, see baa629d

Keep in mind that this only works for JPEGs right now and that this still needs some work if the image is resized / downsampled since the BitmapFactory.Options#inSampleSize value is not correctly handled.

I'm closing this issue for now. Please let us know if this is still an issue.

**Comment by alexmelyon** [comment]
@oprisnik have the same issue
`view.setLayerType(View.LAYER_TYPE_SOFTWARE, null)` not working
I set `setDownsampleEnabled(true)`, otherwise it doesn't call onFinalImageSet at all.
And I cannot get original size. Plz help.
